### PR TITLE
Add entity permalink

### PR DESCRIPTION
**Pull request by wadackel**
## What does this change?

I added to permalink for `RegEntity` viewer display.
I would like to use it when sharing the URL of an item to other developers :dog:

URL format as following:

```
<origin><pathname>?id=<entity id>
```

## References

- [ReactTraining/history: Manage session history with JavaScript](https://github.com/ReactTraining/history)

## Screenshots

demo

## What can I check for bug fixes?

n/a

### CODE REVIEW

File: src/context/HistoryContext.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export type HistoryContextValue = History;
+
+export const HistoryContext = createContext<HistoryContextValue>(history);
+
+export const HistoryContextProvider: React.FC = ({ children }) => (
+  <HistoryContext.Provider value={history}>{children}</HistoryContext.Provider>
+);
+
+export const useHistory = () => {
```
Comment (Quramy) [inline src/context/HistoryContext.tsx]: [IMO] A custom hook should be exposed from `src/hooks` dir.

```ts
/* src/hooks/useHistory.ts */
import { useContext } from 'react';
import { HistoryContext } from '../context/HistoryContext';

export const useHistory = () => {
  return useContext(HistoryContext);
};
```

File: src/containers/viewer/ViewerContainer.ts [hunk 2]
```diff
@@ -110,6 +93,32 @@ export const ViewerContainer = createContainer<ViewerValue>(() => {
     return () => worker.unsubscribe(WorkerEventType.RESULT_CALC, listener);
   }, [worker, current]);
 
+  useEffect(() => {
+    const sync = (location: Location) => {
+      const query = qs.parse(location.search, { decode: false });
```
Comment (Quramy) [inline src/containers/viewer/ViewerContainer.ts]: We can  'query-string' dependency as the following ?

```ts
      const query = new URLSearchParams(location.search);
      const id = query.get('id');
```

Comment (Quramy) [reply src/containers/viewer/ViewerContainer.ts]: See also: https://url.spec.whatwg.org/#interface-urlsearchparams

**Comment by Quramy** [comment]
@wadackel 

I've added some nits comments, BTW I like this feature :smile:

File: src/containers/viewer/ViewerContainer.ts (hunk 2 above)
Comment (wadackel) [reply src/containers/viewer/ViewerContainer.ts]: I've forgotten about `URLSearchParams` :innocent:
I remove `query-string` from dependencies and replace the implementation with it!

Comment (wadackel) [reply src/containers/viewer/ViewerContainer.ts]: Fixed by https://github.com/reg-viz/reg-cli-report-ui/pull/40/commits/fe082fb06171b78ecd9c08d476dda064b72a29cb

File: src/context/HistoryContext.tsx (hunk 1 above)
Comment (wadackel) [reply src/context/HistoryContext.tsx]: Fixed by https://github.com/reg-viz/reg-cli-report-ui/pull/40/commits/cea018a4a6098ed6a55e89ace832966e278eeb7c

**Comment by wadackel** [comment]
@Quramy 

I've fixed the part that you commented on!
thx :dog:

I'm looking forward to the release of this feature :clap:

---

After reviewing this PR, I plan to fix conflicts and Lint.

**Comment by Quramy** [comment]
@wadackel LGTM 👍

**Comment by wadackel** [comment]
Thanks, merged :+1:

**Pull request merged**